Add Grant to the ecosystem of plugins

simov opened the pull request:
:wave: as discussed here https://github.com/fastify/fastify-oauth2/issues/20#issuecomment-685764095 I'm adding [Grant](https://github.com/simov/grant) to the list of available plugins for Fastify.

Grant is an OAuth authentication/authorization plugin that supports both OAuth 2.0 and OAuth 1.0a providers, currently around 200+ that were tested officially, but any other provider implementing the OAuth spec can be configured to work with the plugin.

Grant relies on session to persist state during the OAuth flow:

```js
var fastify = require('fastify')
var cookie = require('fastify-cookie')
var session = require('fastify-session')
var grant = require('fastify-grant')

fastify()
  .register(cookie)
  .register(session, {secret: 'grant', cookie: {secure: false}})
  .register(grant({/*configuration - see below*/}))
```

I published the module as [fastify-grant](https://www.npmjs.com/package/fastify-grant) on NPM, but it points to the main repo since that's the one I'm maintaining, but overall the documentation there is not concerned with a specific handler, and instead it only outlines the configuration needed.

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [X] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

mcollina commented:
https://www.npmjs.com/package/fastify-grant is not really useful

Please point the metadata to its own repo and add a README.

simov commented:
Got it. Added the docs from Grant (some sections were removed), and published a patch on NPM to update the docs there too. Also updated the link in this repo.

mcollina reviewed (approved):
lgtm
